core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -292,54 +286,47 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
    */
   private class Watchers(val key: Any) {
 
-    private[this] val operations = new LinkedList[T]()
+    private[this] var operations = List[T]()
```

Comment:
Supposedly, we could make this @volatile and remove the synchronized on read operations like `watched`. But I think we need to consider whether scala `List` has the right properties for this. It has O(n) `append` and O(n) `size`. O(1) `prepend` though.